### ISSUE TITLE
False-positive removal of "unneeded" parentheses breaks code

### ISSUE THREAD

**Issue by r0skar** [prettier/eslint-plugin-prettier#183]
I am using this great plugin in addition with [eslint-config-prettier](https://github.com/prettier/eslint-config-prettier). Additionally, I use Typescript and Babel with [this plugin](https://github.com/ryansolid/babel-plugin-jsx-dom-expressions). The Babel plugin enables the usage of dynamic expression inside JSX when they are **wrapped in parentheses**.

The problem: prettier doesnt know this and removes them as "unneeded" parentheses. Obviously this breaks my code, because the parentheses are needed for the dynamic expression to work.  I have tried to unset the responsible rule by toggling all config options from the [eslint-config-prettier rules](https://github.com/prettier/eslint-config-prettier/blob/master/index.js), but I just cant turn off the error.

**What version of `eslint` are you using?**
5.16.0

**What version of `prettier` are you using?**
1.17.0

**What version of `eslint-plugin-prettier` are you using?**
3.0.1

**Please paste any applicable config files that you're using (e.g. `.prettierrc` or `.eslintrc` files)**
```js
// eslintrc.js
extends: [
  `eslint:recommended`,
  `plugin:react/recommended`,
  `plugin:@typescript-eslint/recommended`,
  `plugin:prettier/recommended`,
  `prettier/react`,
  `prettier/@typescript-eslint`
]
```
**What source code are you linting?**
```tsx
// index.tsx
return <div>{(state.counter)}</div>
```

**What did you expect to happen?**
Parentheses around `state.count` should be left in tact.

**What actually happened?**
`Replace '(state.counter)' with 'state.counter' eslint(prettier/prettier)`

**Comment by BPScott** [comment]
`eslint-plugin-prettier` has no opinions on code formatting - it is merely a bridge between `eslint` and `prettier`, it just does whatever `prettier` tells it to. If you're having issues with prettier formatting please raise an issue in the [prettier repository](https://github.com/prettier/prettier).
